BitwardenShared/Core/Vault/Services/VaultTimeoutService.swift

```diff
@@ -171,12 +200,43 @@ class DefaultVaultTimeoutService: VaultTimeoutService {
         }
     }
 
+    func sessionTimeoutAction(userId: String?) async throws -> SessionTimeoutAction {
+        let hasMasterPassword = try await stateService.getUserHasMasterPassword(userId: userId)
+        let timeoutAction = try await stateService.getTimeoutAction(userId: userId)
+        guard hasMasterPassword else {
+            let isBiometricsEnabled = try await biometricsRepository.getBiometricUnlockStatus().isEnabled
+            let isPinEnabled = try await isPinUnlockAvailable(userId: userId)
+            if isPinEnabled || isBiometricsEnabled {
+                return timeoutAction
+            } else {
+                // If the user doesn't have a master password and hasn't enabled a pin or
+                // biometrics, their timeout action needs to be logout.
+                return .logout
+            }
+        }
+        return timeoutAction
+    }
+
     func setLastActiveTime(userId: String) async throws {
-        try await stateService.setLastActiveTime(timeProvider.presentTime, userId: userId)
+        let now = timeProvider.presentTime
+        try await stateService.setLastActiveTime(now, userId: userId)
+        let vaultTimeout = try await sessionTimeoutValue(userId: userId)
+        try await updateSharedTimeout(
+            lastActiveTime: now,
+            timeoutValue: vaultTimeout,
+            userId: userId
+        )
     }
 
     func setVaultTimeout(value: SessionTimeoutValue, userId: String?) async throws {
         try await stateService.setVaultTimeout(value: value, userId: userId)
+        guard let userId else { return }
+        let lastActiveTime = try await stateService.getLastActiveTime(userId: userId)
+        try await updateSharedTimeout(
+            lastActiveTime: lastActiveTime,
+            timeoutValue: value,
+            userId: userId
+        )
```

Comment:
⛏️ Tests are missing for all the scenarios where `try await updateSharedTimeout(...)` can throw.